src/lib.rs

```diff
@@ -693,14 +703,30 @@ impl CacheEntry {
 /// A timing value which may be absent or present
 ///
 /// Defaults to -1 in the absent case.
-#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[derive(Serialize, PartialEq, Debug)]
 #[serde(rename_all = "camelCase")]
 #[serde(untagged)]
 pub enum OptionalTiming {
     TimedContent(u32),
     NotApplicable
 }
 
+impl Deserialize for OptionalTiming {
+    fn deserialize<D>(deserializer: D) -> Result<OptionalTiming, D::Error>
+        where D: Deserializer
+    {
+        let deser_result: serde_json::Value = try!(serde::Deserialize::deserialize(deserializer));
+        match deser_result {
+            //serde_json::Value::Number(n) => Ok(OptionalTiming::TimedContent(n.as_u64().unwrap() as u32)),
+            serde_json::Value::Number(ref n) if n.as_i64().unwrap() >= 0 as i64 => 
```

Comment:
Will this panic if there is non-i64 content like `5.3` in the JSON for this field? Also, the `as i64` cast shouldn't be necessary here.